src/clj/puppetlabs/services/ca/certificate_authority_core.clj

```diff
@@ -95,6 +95,16 @@
           (rr/status 304)
           (rr/content-type "text/plain")))))
 
+(schema/defn handle-put-certificate-revocation-list!
+  [upstream-crl-pem :- InputStream
+   {:keys [cacrl infra-crl-path enable-infra-crl]} :- ca/CaSettings]
+  (ca/update-crl upstream-crl-pem cacrl)
```

Comment:
I think we want to put wrap this in a lock

Reply:
yeah, haven't gotten to adding the locking yet, but it's on my list